storm/check.py

```diff
@@ -0,0 +1,688 @@
+                                    (component_value, ks_tags))
+                    r[k].append(k_stat)
+            return r
+
+    def report_gauge(self, metric, value, tags, additional_tags=list()):
+        """ Report the Gauge Metric.
+
+        :param metric:
+        :param value:
+        :param tags:
+        :param additional_tags:
+        :return:
+        """
+        all_tags = tags + [
+            'env:{}'.format(self.environment_name),
+            'environment:{}'.format(self.environment_name)] + additional_tags
+        self.gauge(
+            metric=metric,
+            value=value,
+            tags=all_tags
+        )
+
+    def report_histogram(self, metric, value, tags, additional_tags=list()):
+        """ Report the Histogram Metric.
+
+        :param metric:
+        :param value:
+        :param tags:
+        :param additional_tags:
+        :return:
+        """
+        all_tags = tags + [
+            'env:{}'.format(self.environment_name),
+            'environment:{}'.format(self.environment_name)] + additional_tags
+        self.histogram(
+            metric=metric,
+            value=value,
+            tags=all_tags
+        )
+
+    def update_from_config(self, instance):
+        """ Update Configuration tunables from instance configuration.
+
+        :param instance: Agent config instance.
+        :return: None
+        """
+        if instance.get('https', self.init_config.get('https', "False")).lower() in ['true', 't', '1']:
+            self.http_prefix = 'https://'
+        else:
+            self.http_prefix = 'http://'
+        self.nimbus_server = instance.get('server', self.init_config.get('server', 'localhost:9005'))
+        self.environment_name = instance.get('environment', self.init_config.get('environment', 'dev'))
+        self.additional_tags = instance.get('tags', [])
+        self.excluded_topologies = instance.get('excluded', [])
+        intervals = instance.get('intervals', self.init_config.get('intervals', [60]))
+        self.intervals = []
+        if isinstance(intervals, (list, tuple)):
+            for interval in intervals:
+                if isinstance(interval, (list, tuple)):
+                    for i in interval:
+                        self.intervals.append(int(i))
+                else:
+                    self.intervals.append(int(interval))
+        elif isinstance(intervals, (int, long)):
+            self.intervals.append(intervals)
+        else:
+            self.intervals.extend([int(i) for i in intervals.split(',')])
+
+    def check(self, instance):
+        """ Perform the agent check.
+
+        :param instance: Agent instance.
+        :return: None
+        """
+        # Setup
+        self.update_from_config(instance)
+
+        # Cluster Stats
+        cluster_stats = self.get_storm_cluster_summary()
+        for k, v in self.process_cluster_stats(self.environment_name, cluster_stats).items():
```

Comment:
Each `process_*` function could be in charge of sending its metrics instead of returning a dict and looping through it. This would also avoid the need for `_gt` to return a tuple with the value and the tags in it.